action/protocol/rewarding/fund.go

```diff
@@ -119,6 +119,73 @@ func (p *Protocol) deposit(
 	}, nil
 }
 
+func (p *Protocol) depositAndBurn(
+	ctx context.Context,
+	sm protocol.StateManager,
+	burnAddr address.Address,
+	amount, burnAmount *big.Int,
+	transactionLogType iotextypes.TransactionLogType,
+) ([]*action.TransactionLog, error) {
+	var (
+		actionCtx           = protocol.MustGetActionCtx(ctx)
+		fCtx                = protocol.MustGetFeatureCtx(ctx)
+		accountCreationOpts = []state.AccountCreationOption{}
+	)
+	if fCtx.CreateLegacyNonceAccount {
+		accountCreationOpts = append(accountCreationOpts, state.LegacyNonceAccountTypeOption())
+	}
+	// Subtract balance from caller
+	acc, err := accountutil.LoadAccount(sm, actionCtx.Caller, accountCreationOpts...)
+	if err != nil {
+		return nil, err
+	}
+	if err := acc.SubBalance(amount); err != nil {
+		return nil, err
+	}
+	if !isZero(burnAmount) {
+		if err := acc.SubBalance(burnAmount); err != nil {
+			return nil, err
+		}
+	}
+	if err := accountutil.StoreAccount(sm, actionCtx.Caller, acc); err != nil {
+		return nil, err
+	}
+	// Add balance to fund
+	var (
+		f    = fund{}
+		tLog = []*action.TransactionLog{
+			&action.TransactionLog{
+				Type:      transactionLogType,
+				Sender:    actionCtx.Caller.String(),
+				Recipient: address.RewardingPoolAddr,
```

Comment:
to be consistent with `burn`, `rewardAddr` could be introduced to this function as well.

Reply:
`rewardAddr` is a special address, reward pool balance is a regular address (but a special k/v pair), the gas fee is not added to the pool via a regular transfer, so I think it's OK to directly use the special address.

In other words, you can only pass in the special value `address.RewardingPoolAddr` for this param